fhir/com.b2international.snowowl.fhir.core/src/com/b2international/snowowl/fhir/core/request/bundle/FhirBundleSearchRequest.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2021 B2i Healthcare Pte Ltd, http://b2i.sg
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.b2international.snowowl.fhir.core.request.bundle;
+
+import java.io.IOException;
+import java.util.UUID;
+
+import com.b2international.commons.CompareUtils;
+import com.b2international.snowowl.core.domain.RepositoryContext;
+import com.b2international.snowowl.core.request.SearchResourceRequest;
+import com.b2international.snowowl.fhir.core.codesystems.BundleType;
+import com.b2international.snowowl.fhir.core.model.Bundle;
+import com.b2international.snowowl.fhir.core.model.Bundle.Builder;
+import com.b2international.snowowl.fhir.core.model.Meta;
+import com.b2international.snowowl.fhir.core.model.dt.Coding;
+import com.b2international.snowowl.fhir.core.model.dt.Instant;
+
+/**
+ * @since 8.0.0
+ */
+public class FhirBundleSearchRequest extends SearchResourceRequest<RepositoryContext, Bundle> {
+
+	private static final long serialVersionUID = 1L;
+
+	@Override
+	protected Bundle createEmptyResult(int limit) {
+		return prepareBundle().total(0).build();
+	}
+
+	@Override
+	protected Bundle doExecute(RepositoryContext context) throws IOException {
+		throw new UnsupportedOperationException();
```

Comment:
You can search for bundles using the ResourceSearchRequest API and filter for the `bundles` resource type.